R/safe2x2Test.R

```diff
@@ -1,1148 +1,2016 @@
+  }
 
-    mean.s.os <- numeric(nrow(H1.deltamin))
+  if(length(aSample) != length(bSample)){
+    stop("Can only process complete data blocks: provide vectors with numbers of positive observations per timepoint, see example in helpfile.")
+  }
 
-    set.seed(seed.time)
+  if(any(aSample > na | aSample < 0) | any(bSample > nb | bSample < 0)){
+    stop("Provided sample sizes within blocks, na and nb, do not match provided aSample and bSample.")
+  }
 
-    for (h in 1:nrow(H1.deltamin)) {
-      theta.a <- H1.deltamin[h, 1]
-      theta.b <- H1.deltamin[h, 2]
+  #unpack the prior values
+  betaA1 <- priorValues[["betaA1"]]
+  betaA2 <- priorValues[["betaA2"]]
+  betaB1 <- priorValues[["betaB1"]]
+  betaB2 <- priorValues[["betaB2"]]
+
+  #set starting E variable
+  if(restriction == "difference"){
+    EVariable <- createStartEWithRestriction(na = na, nb = nb,
+                                             delta = delta,
+                                             logOdds = FALSE,
+                                             betaA1 = betaA1, betaA2 = betaA2,
+                                             gridSize = gridSize
+                                             )
+
+  } else if(restriction == "logOddsRatio"){
+    EVariable <- createStartEWithRestriction(na = na, nb = nb,
+                                             delta = delta,
+                                             logOdds = TRUE,
+                                             betaA1 = betaA1, betaA2 = betaA2,
+                                             gridSize = gridSize
+                                             )
+  } else if(restriction == "none"){
+    EVariable <- updateE(totalSuccessA = 0, totalFailA = 0,
+                         totalSuccessB = 0, totalFailB = 0,
+                         na = na, nb = nb,
+                         betaA1 = betaA1, betaA2 = betaA2,
+                         betaB1 = betaB1, betaB2 = betaB2)
+
+    #for updating without restriction, use totals: store them here
+    totalSuccessA <- cumsum(aSample)
+    totalSuccessB <- cumsum(bSample)
+    groupSizeVecA <- seq_along(totalSuccessA)*na
+    groupSizeVecB <- seq_along(totalSuccessB)*nb
+    totalFailA <- groupSizeVecA - totalSuccessA
+    totalFailB <- groupSizeVecB - totalSuccessB
+  }
 
-      rejected <- logical(M)
-      n.final <- numeric(M)
+  currentE <- 1
+  for(i in seq_along(aSample)){
+    #use only new data to calculate the new E variable
+    newE <- calculateE(na1 = aSample[i],
+                       na = na,
+                       nb1 = bSample[i],
+                       nb = nb,
+                       thetaA = EVariable[["thetaA"]],
+                       thetaB = EVariable[["thetaB"]],
+                       theta0 = EVariable[["theta0"]])
+    currentE <- newE * currentE
+
+    #in simulation setting, only interested in the stopping time
+    if(simSetting & currentE >= (1/alphaSim)){
+      break
```

Comment:
wellicht goed om `break()` te schrijven hier.

Reply:
zou je kunnen uitleggen waarom? break is standaard voor control flow, wat is volgens jou het voordeel van break met brackets?

Reply:
De haakjes maken het duidelijker dat het een functie is. Beide werken, dus daarom heb ik een voorkeur voor met haakjes.